packages/components/inputs/date-time-input/src/date-time-input.tsx

```diff
@@ -39,99 +46,137 @@ const activationTypes = [
   Downshift.stateChangeTypes.clickItem,
 ];
 
-const preventDownshiftDefault = (event) => {
-  // eslint-disable-next-line no-param-reassign
+type TPreventDownshiftDefaultEvent = {
+  relatedTarget?: unknown;
+  nativeEvent?: {
+    preventDownshiftDefault?: boolean;
+  };
+} & FocusEvent<HTMLButtonElement> &
+  KeyboardEvent<HTMLInputElement | HTMLButtonElement>;
+
+const preventDownshiftDefault = (event: TPreventDownshiftDefaultEvent) => {
   event.nativeEvent.preventDownshiftDefault = true;
 };
 
 // This keeps the menu open when the user focuses the time input (thereby
 // blurring the regular input/toggle button)
-const createBlurHandler = (timeInputRef) => (event) => {
-  event.persist();
-  if (event.relatedTarget === timeInputRef.current) {
-    preventDownshiftDefault(event);
-  }
+const createBlurHandler =
+  (timeInputRef: RefObject<HTMLInputElement>) =>
+  (event: TPreventDownshiftDefaultEvent) => {
+    event.persist();
+    if (event.relatedTarget === timeInputRef.current) {
+      preventDownshiftDefault(event);
+    }
+  };
+
+type TEvent = {
+  target: {
+    id?: string;
+    name?: string;
+    value?: string;
+  };
+};
+
+type TPlaceholder = {
+  id: string;
+  description: string;
+  defaultMessage: string;
 };
 
-class DateTimeInput extends Component {
+type TIntl = {
+  locale: string;
+  formatMessage: (placeholder: TPlaceholder) => string;
+};
+
+export type TDateTimeInputProps = {
+  intl: TIntl;
+
+  /**
+   * Horizontal size limit of the input field.
+   */
+  horizontalConstraint?:
+    | 6
+    | 7
+    | 8
+    | 9
+    | 10
+    | 11
+    | 12
+    | 13
+    | 14
+    | 15
+    | 16
+    | 'scale'
+    | 'auto';
+  /**
+   * The selected date, must either be an empty string or a date formatted in ISO 8601 (e.g. "2018-10-04T09:00:00.000Z").
+   */
+  value: string;
+  /**
+   * Called when the date changes. Called with an event containing an empty string (no value) or a string in this format: "YYYY-MM-DD".
+   * <br />
+   * Signature: `(event) => void`
+   */
+  onChange: (event: TEvent) => void;
+  /**
+   * Called when the date input gains focus.
+   */
+  onFocus?: FocusEventHandler;
+  /**
+   * Called when the date input loses focus.
+   */
+  onBlur?: (event: TEvent) => void;
+  /**
+   * Specifies the time zone in which the calendar and selected values are shown. It also influences how entered dates and times are parsed.
```

Comment:
Maybe we could add an example of the time zone possible value to better understand its format.